### PR TITLE
snapshotter: don't emit `null` snapshot files when no snapshots are r…

### PR DESCRIPTION
**Pull request by er9781**
Previously if you initalized snapshotter in a test & invoked Verify()
at the end, if no snapshots were recorded, a `null` file would be
output. This was clutter for folks to avoid adding to git.

Here we simply don't write the file if we see no snapshots recorded during the test.

### CODE REVIEW

File: snapshotter/snapshotter.go [hunk 1]
```diff
@@ -87,6 +87,17 @@ func (s *Snapshotter) Verify() {
 	}
 	name := filepath.Join("testdata", strings.Replace(strings.Replace(s.t.Name(), "/", "-", -1), ":", "-", -1)+nameSuffix+".snapshots.json")
 	if *rewrite {
+		// If there are no snapshots, then when rewriting, we want to remove the file if it exists.
+		if len(s.snapshots) == 0 {
+			if _, err := os.Stat(name); !os.IsNotExist(err) {
+				// the file exists, so let's remove it.
+				err := os.Remove(name)
+				if err != nil {
+					panic(err)
```
Comment (ChenJesse) [inline snapshotter/snapshotter.go]: prob want to use `s.t.Errorf`? that way if you use snapshotter in unit tests, you can pass `testing.T` in and let it handle whatever happened

File: snapshotter/snapshotter.go [hunk 2]
```diff
@@ -87,6 +87,17 @@ func (s *Snapshotter) Verify() {
 	}
 	name := filepath.Join("testdata", strings.Replace(strings.Replace(s.t.Name(), "/", "-", -1), ":", "-", -1)+nameSuffix+".snapshots.json")
 	if *rewrite {
+		// If there are no snapshots, then when rewriting, we want to remove the file if it exists.
+		if len(s.snapshots) == 0 {
+			if _, err := os.Stat(name); !os.IsNotExist(err) {
+				// the file exists, so let's remove it.
```
Comment (ChenJesse) [inline snapshotter/snapshotter.go]: nit - capitalize sentences 🙈

File: snapshotter/snapshotter.go (hunk 1 above)
Comment (ChenJesse) [reply snapshotter/snapshotter.go]: being really nitty here, but slight pref for 
```
if err := os.Remove(name); err != nil {

}
```
to keep things more compact

**Review by ChenJesse — COMMENTED** [review]
nit - commit message too long? and also small typo (extra 'd' at end of "recorded")

mostly nits, but this change is great!

File: snapshotter/snapshotter.go (hunk 1 above)
Comment (er9781) [reply snapshotter/snapshotter.go]: I mean, can we differentiate the file didn't exist vs other os remove error?

I swapped it to early return in samsara's conventional style for clarity, but still checked for path exists first.

Comment (ChenJesse) [reply snapshotter/snapshotter.go]: ok that looks a lot cleaner, thanks

not sure if theres miscommunication but what i meant was 
```
// The file exists, so let's remove it.
err := os.Remove(name)
if err != nil {
    s.t.Errorf("failed to remove the existing snapshot file %s", name)
}
```

to 

```
// The file exists, so let's remove it.
if err := os.Remove(name); err != nil {
    s.t.Errorf("failed to remove the existing snapshot file %s", name)
}
```

**Review by ChenJesse — COMMENTED** [review]
may be nice to write a quick test?

**Comment by er9781** [comment]
> may be nice to write a quick test?

@ChenJesse yeah, I initially added a test, but found out there's one already one [here](https://github.com/samsarahq/go/blob/13570df44b46ea943b9e7e2b16fd1993542e01e3/snapshotter/snapshotter_test.go#L46-L49). I ran it locally and it didn't produce a snapshot file (and it did without my change), which makes me confident in the change.

I didn't add a test for snapshot file -> removes snapshot file, but that would involve more testing infra and doesn't seem worth it.

File: snapshotter/snapshotter.go (hunk 1 above)
Comment (er9781) [reply snapshotter/snapshotter.go]: Ohhhh, yeah, np. Thought you were advocating not checking for existence and just relying on the error from remove.

Comment (er9781) [reply snapshotter/snapshotter.go]: made the change.

**Review by aws5295 — APPROVED** [review]
👍

**Pull request merged**